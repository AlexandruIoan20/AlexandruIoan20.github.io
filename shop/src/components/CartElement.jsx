import React from "react";

const CartElement = (props) => { 
    const { item, onIncreasingQ, onInputButton } = props;
    return (
        <section className="cart-element">
            <p> { `${item.name}` }</p>
            <img src= { item.url } width = "100px" height="100px" alt="...loading" />

            <article className="price-q-informator">
                <form> 
                    <button className="cart-button" onClick={ (e) => 
                        {
                            e.preventDefault ();
                            onInputButton("-", item)
                        }
                        }> - </button>
                    <input type="number" 
                        min= {1} 
                        step = '1' 
                        onChange = { (e) => { onIncreasingQ(item, e.target.value)}}
                        value = { item.quantity} />
                    <button className="cart-button" onClick = { (e) => 
                        {
                            e.preventDefault();
                            onInputButton("+", item)
                        }
                        }> + </button>
                </form>
                <p className="price"> {`${item.quantity} x ${item.price * item.quantity}$ / ${item.quantity} / ${item.price}$`} </p>
            </article>
        </section>
    )
};

export default CartElement;