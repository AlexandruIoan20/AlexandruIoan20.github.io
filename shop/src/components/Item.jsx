import React, { useState, useEffect } from "react";
import { useLocation, NavLink  } from "react-router-dom";
import { useCartUpdate, useCartItems } from "./CartItemsProvider";

const Item = (props) => { 
    const location = useLocation();
    const updateCart = useCartUpdate();
    const itemsInCart = useCartItems();
    console.log(props, "props");
    console.log(location, "location");

    const item = location.state.item;

    return ( 
        <section className="item-holder">
            <figure className="item">
                <img src= {item.url} width = "300px" height = "450px" alt="loading" />
                <h2 role="h2"> { item.name } </h2>
                <NavLink
                    state = { { item: item }}
                    to = "/cart"><button onClick = { () => { updateCart(item);}}> Add to cart </button></NavLink>

                <p role="p"> { `${item.price}$`  }</p>
            </figure>
        </section>
    )
};

export default Item;