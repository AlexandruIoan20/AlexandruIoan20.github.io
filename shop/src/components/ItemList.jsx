import React, { useState, useEffect } from "react";
import { useParams, NavLink } from "react-router-dom";
import  { createApi } from 'unsplash-js';
import ITEMS from "./Helpers/items";
import { useNavigate } from "react-router-dom";

const unsplash = createApi({
    accessKey: 'znEgxfmnj804269_4rFVRaNx48Y_aCjmL9xzsh133Cw'
  });

const ItemList = () => { 
    const navigate = useNavigate();
    const urlID = useParams();
    const [items, setItems] = useState([]);
    
    async function buildItemList () { 
        let list = {};

        for(let i = 0; i < ITEMS.length; i++) {
            if(ITEMS[i].category === urlID.id) { 
                list = ITEMS[i].list
            }
        }

        let fetchedElements = [];
        await unsplash.search.getPhotos({ 
            query: urlID.id,
            page: 10,
            perPage: 10,
            orientation: 'portrait',
        }).then(arr => { 
            for(let i = 0; i < 6; i++) { 
                fetchedElements[i] = { 
                    name: list[i].name, 
                    price: list[i].price,
                    url: arr.response.results[i].urls.small_s3,
                    quantity: 1,
                    id: arr.response.results[i].blur_hash,
                }
            };
        }).then( () => { 
            setItems(fetchedElements);
        })
    }

    useEffect( () => { 
        if(items.length === 0) { 
            navigate("/*")
         }
        buildItemList();
     }, [])

    return ( 
        <section className="items-informations">
            <p className="item-list-title">We found {`${items.length}`} items.</p>
            <div className="items-list">
                {items.map(item => { 
                    return ( 
                        <NavLink key = { item.id }  to = {`/categories/${urlID.id}/${item.name}`}
                            state={{ item: item }}>
                            <article className="item-in-list">
                                <img src={ item.url } width = "200px" height = "250px" alt="loading..." />
                                <p className="item-name"> { item.name }</p>
                                <p> { `${item.price}$` }</p>
                            </article>
                        </NavLink>
                    )
                })} 
            </div>
        </section> )

    
};

export default ItemList;