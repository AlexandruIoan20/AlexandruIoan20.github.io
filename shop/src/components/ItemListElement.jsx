import React from "react";

const ItemListElement = () => { 
    return ( 
        <figure className="item-list-element">
            <img src="itemListElement" alt="loading" />
            <button>Item Name</button>
        </figure>
    )
};

export default ItemListElement;