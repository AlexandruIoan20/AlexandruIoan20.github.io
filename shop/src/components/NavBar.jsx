import React from "react";
import SearchBar from "./SearchBar";
import  { NavLink }from "react-router-dom";

const NavBar = () => { 
    return (
        <section className="navbar">
            <article className="navbar-buttons">
                <button>Home</button>
                <button>Cart</button>
            </article>
        <SearchBar />
        </section>
    )
};

export default NavBar; 