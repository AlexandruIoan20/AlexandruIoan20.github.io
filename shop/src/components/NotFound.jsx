import React from "react";
import { useNavigate } from "react-router";

const NotFound = () => { 
    const navigate = useNavigate();

/*     setTimeout( () => { 
        navigate("/")
    }, 2000); */
    return( 
        <section className="not-found-page">
            <article className="not-found-elements">
                <h1 className="not-found">Not Found</h1>
                <button className="start-button" onClick={ () => {navigate("/")}}>Go Home</button>
            </article>
        </section>
    )
};

export default NotFound;