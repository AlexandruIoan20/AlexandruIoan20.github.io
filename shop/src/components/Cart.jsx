import React from "react";
import CartElement from "./CartElement";

const Cart = () => { 
    const elements = [
        { 
            name: "Phone", 
            value: 3,
        },
        { 
            name: "T-shirt",
            value: 2 
        }
    ]
    return (
        <section className="cart">
            {elements.map(element => { 
                return <CartElement/>
            })}
        </section>
    )
};

export default Cart; 