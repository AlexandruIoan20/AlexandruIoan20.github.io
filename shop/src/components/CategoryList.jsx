import React, { useState, useEffect } from "react";
import Category from "./Category";
import CATEGORIES from './Helpers/CategoriesArray.js';
import ItemList from "./ItemList";
import { NavLink } from "react-router-dom";
import  { createApi } from 'unsplash-js';
import CartItemsProvider from "./CartItemsProvider";


const unsplash = createApi({
    accessKey: 'znEgxfmnj804269_4rFVRaNx48Y_aCjmL9xzsh133Cw'
  });

async function getState ( getStateArray ) { 
    for(let i = 0; i < CATEGORIES.length; i++) {
        let images = [];
        await unsplash.search.getPhotos({ 
            query: CATEGORIES[i].use,
            page: 1,
            perPage: 1,
            orientation: 'portrait',
        }).then(arr => { 
            images[i] = arr.response.results[0].urls.small_s3;
        }).then( () => { getStateArray[i] = images[i]} );
    }
}

const CategoryList = () => { 
    const [categories, setCategories] = useState ([]);

    async function createCategories () { 
        let catState = [], array = [];
        await getState(catState).then(() => {
            for(let i = 0; i < CATEGORIES.length; i++) {
                array.push( { name: CATEGORIES[i].view, query:CATEGORIES[i].use, id: CATEGORIES[i].use, url: catState[i]});
                setCategories(array);
            }
        })
    };

    useEffect ( () => { 
        createCategories();
    }, []);

    return(
        <section className="category-list">
            {categories.map(category => { 
                return ( 
                    <article key = { category.id }>
                        <NavLink
                            key = { categories.id } to = {`/categories/${category.id}`}>
                            <Category category = { category }/></NavLink>
                    </article>
                )
            })}
        </section>
    )
};

export default CategoryList;
