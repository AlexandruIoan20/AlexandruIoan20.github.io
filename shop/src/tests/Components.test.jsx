import React from "react";
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Item from "../components/Item";
import ItemList from "../components/ItemList"
import SearchBar from "../components/SearchBar";
import NavBar from "../components/NavBar";
import CategoryList from "../components/CategoryList";
import ItemListElement from "../components/ItemListElement";
import Cart from "../components/Cart";
import CartElement from "../components/CartElement";
import NotFound from "../components/NotFound";

describe("Plain components", () => { 
    it("renders Item", () => { 
        const { container } = render(<Item/>);
        expect(container).toMatchSnapshot();
    });

    it("renders an item in an item list", () => { 
        const { container } = render(<ItemList/>)
        expect (container).toMatchSnapshot();
    }); 

    it("renders the searchbar", () => { 
        const { container } = render(<SearchBar />);
        expect(container).toMatchSnapshot();
    })

    it("renders the navbar", () => { 
        const { container } = render(<NavBar />);
        expect (container).toMatchSnapshot();
    });

    it("renders category list", () => { 
        const {container } = render(<CategoryList /> );
        expect(container).toMatchSnapshot();
    });

    it("renders item list element", () => { 
        const { container } = render(<ItemListElement />);
        expect(container).toMatchSnapshot();
    });

    it("renders cart element", () => { 
        const { container } = render(<CartElement />);
        expect(container).toMatchSnapshot();
    })
    
    it("renders the card", () => { 
        const { container } = render(<Cart/>);
        expect(container).toMatchSnapshot();
    });

    it("render not found", () => { 
        const { container } = render(<NotFound/>);
        expect(container).toMatchSnapshot();
    })
})