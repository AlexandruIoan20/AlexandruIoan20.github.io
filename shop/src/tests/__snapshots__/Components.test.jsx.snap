// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plain components render not found 1`] = `
<div>
  <h1>
    Not Found
  </h1>
</div>
`;

exports[`Plain components renders Item 1`] = `
<div>
  <figure
    class="item"
  >
    <img
      alt="loading"
      src="image"
    />
    <h2
      role="h2"
    >
      title
    </h2>
    <h3
      role="h3"
    >
      category
    </h3>
    <button>
       Add to cart 
    </button>
    <p
      role="p"
    >
      description
    </p>
  </figure>
</div>
`;

exports[`Plain components renders an item in an item list 1`] = `
<div>
  <section
    class="item-list"
  >
    <p
      class="item-list-title"
    >
      We found 
      1
       items.
    </p>
    <figure
      class="item-list-element"
    >
      <img
        alt="loading"
        src="itemListElement"
      />
      <button>
        Item Name
      </button>
    </figure>
  </section>
</div>
`;

exports[`Plain components renders cart element 1`] = `
<div>
  <section
    class="cart-element"
  >
    <p>
      Product Name
    </p>
    <form>
      <button
        class="cart-button"
      >
         - 
      </button>
      <input
        min="1"
        step="1"
        type="number"
      />
      <button
        class="cart-button"
      >
         + 
      </button>
    </form>
  </section>
</div>
`;

exports[`Plain components renders category list 1`] = `
<div>
  <section
    class="category-list"
  >
    <figure>
      <img
        alt="loading"
        src="category"
      />
      <h2>
         Category 
      </h2>
    </figure>
    <figure>
      <img
        alt="loading"
        src="category"
      />
      <h2>
         Category 
      </h2>
    </figure>
    <figure>
      <img
        alt="loading"
        src="category"
      />
      <h2>
         Category 
      </h2>
    </figure>
  </section>
</div>
`;

exports[`Plain components renders item list element 1`] = `
<div>
  <figure
    class="item-list-element"
  >
    <img
      alt="loading"
      src="itemListElement"
    />
    <button>
      Item Name
    </button>
  </figure>
</div>
`;

exports[`Plain components renders the card 1`] = `
<div>
  <section
    class="cart"
  >
    <section
      class="cart-element"
    >
      <p>
        Product Name
      </p>
      <form>
        <button
          class="cart-button"
        >
           - 
        </button>
        <input
          min="1"
          step="1"
          type="number"
        />
        <button
          class="cart-button"
        >
           + 
        </button>
      </form>
    </section>
    <section
      class="cart-element"
    >
      <p>
        Product Name
      </p>
      <form>
        <button
          class="cart-button"
        >
           - 
        </button>
        <input
          min="1"
          step="1"
          type="number"
        />
        <button
          class="cart-button"
        >
           + 
        </button>
      </form>
    </section>
  </section>
</div>
`;

exports[`Plain components renders the navbar 1`] = `
<div>
  <section
    class="navbar"
  >
    <article
      class="navbar-buttons"
    >
      <button>
        Home
      </button>
      <button>
        Cart
      </button>
    </article>
    <form
      class="search-bar"
    >
      <input
        autocomplete="off"
        placeholder="...your product"
        type="text"
      />
      <button>
        Search
      </button>
    </form>
  </section>
</div>
`;

exports[`Plain components renders the searchbar 1`] = `
<div>
  <form
    class="search-bar"
  >
    <input
      autocomplete="off"
      placeholder="...your product"
      type="text"
    />
    <button>
      Search
    </button>
  </form>
</div>
`;
